python/eight_mile/pytorch/embeddings.py

```diff
@@ -164,6 +165,41 @@ def get_vsz(self):
         return self.embed.get_vsz()
 
 
+class CharTransformerEmbeddings(PyTorchEmbeddings):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.embed = LookupTableEmbeddings(**kwargs)
+        self.d_model = kwargs.get('wsz', 30)
+        self.num_heads = kwargs.get('num_heads', 3)
+        self.rpr_k = kwargs.get('rpr_k', 10)
+        layers = kwargs.get('layers', 1)
+        pdrop = kwargs.get('pdrop', 0.5)
+        self.char_comp = TransformerEncoderStackWithLengths(self.num_heads, self.d_model, pdrop, False, layers,
+                                                            rpr_k=self.rpr_k, input_sz=self.embed.output_dim)
+
+    def forward(self, xch):
+        B, T, W = xch.shape
+        flat_chars = xch.view(-1, W)
+        char_embeds = self.embed(flat_chars)
+
+        # Calculate the lengths of each word
+        lengths = torch.sum(flat_chars != Offsets.PAD, dim=1)
+        patched_lengths = lengths.masked_fill(lengths == 0, 1)
```

Comment:
Do we need to patch the lengths? This is done in the LSTM to avoid cuda errors, Does the transformer error with a length of `0`? It seems like it should be able to handle it, the mask will just have a row of all zeros which seems fine